opentelemetry-api/src/opentelemetry/context/__init__.py

```diff
@@ -104,16 +86,29 @@ def get_current() -> Context:
     return _RUNTIME_CONTEXT.get_current()  # type:ignore
 
 
-def set_current(context: Context) -> Context:
-    """To associate a context with program execution, the Context
-    API provides a function which takes a Context.
+def attach(context: Context) -> object:
+    """Associates a Context with the caller's current execution unit. Returns
+    a token that can be used to restore the previous Context.
+
+    Args:
+        context: The Context to set as current.
+    """
+    get_current()
+
+    return _RUNTIME_CONTEXT.set_current(context)  # type:ignore
```

Comment:
How about renaming `_RUNTIME_CONTEXT.set_current` to `_RUNTIME_CONTEXT.attach` for consistency?

Reply:
updated